Alerting docs: update Alerting Provisioning (#83376)
grafana/grafana: docs/sources/alerting/set-up/provision-alerting-resources/file-provisioning/index.md

{{< admonition type="note" >}}

- Provisioning with configuration files is not available in Grafana Cloud.

- You cannot edit provisioned resources from files in Grafana. You can only change the resource properties by changing the provisioning file and restarting Grafana or carrying out a hot reload. This prevents changes being made to the resource that would be overwritten if a file is provisioned again or a hot reload is carried out.

- Provisioning via configuration files takes place during the initial set up of your Grafana system, but you can re-run it at any time using the [Grafana Admin API](/docs/grafana/<GRAFANA_VERSION>/developers/http_api/admin#reload-provisioning-configurations).

- Importing an existing alerting resource results in a conflict. First, when present, remove the resources you plan to import.
  {{< /admonition >}}

## Import alert rules

Create or delete alert rules via provisioning files in your Grafana instance(s).

1. Find the alert rule group in Grafana.
1. [Export][alerting_export] and download a provisioning file for your alert rules.

> 1. Copy the contents into a YAML or JSON configuration file and place it in the `provisioning/alerting` directory of the Grafana instance you wish to import the alerting resources.

Reviewer: ```suggestion
1. Copy the contents into a YAML or JSON configuration file and add it to the `provisioning/alerting` directory of the Grafana instance you want to import the alerting resources to.
```

   Example configuration files can be found below.

> 1. Restart your Grafana instance (or reload the provisioned files via the Admin API).

Reviewer: ```suggestion
1. Restart your Grafana instance (or reload the provisioned files using the Admin API).
```

## Import contact points

> Create or delete contact points via provisioning files in your Grafana instance(s).

Reviewer: ```suggestion
Create or delete contact points using provisioning files in your Grafana instance(s).
```

1. Find the contact point in Grafana.
1. [Export][alerting_export] and download a provisioning file for your contact point.

> 1. Copy the contents into a YAML or JSON configuration file and place it in the `provisioning/alerting` directory of the Grafana instance you wish to import the alerting resources.

Reviewer: ```suggestion
1. Copy the contents into a YAML or JSON configuration file and add it to the `provisioning/alerting` directory of the Grafana instance you want to import the alerting resources to.
```

   Example configuration files can be found below.

> 1. Restart your Grafana instance (or reload the provisioned files via the Admin API).

Reviewer: ```suggestion
1. Restart your Grafana instance (or reload the provisioned files using the Admin API).
```

{{< /collapse >}}

## Import templates

> Create or delete templates via provisioning files in your Grafana instance(s).

Reviewer: ```suggestion
Create or delete templates using provisioning files in your Grafana instance(s).
```

1. Find the notification template in Grafana.
1. [Export][alerting_export] a template by copying the template content and title.

> 1. Copy the contents into a YAML or JSON configuration file and place it in the `provisioning/alerting` directory of the Grafana instance you wish to import the alerting resources.

Reviewer: ```suggestion
1. Copy the contents into a YAML or JSON configuration file and add it to the `provisioning/alerting` directory of the Grafana instance you want to import the alerting resources to.
```

   Example configuration files can be found below.

> 1. Restart your Grafana instance (or reload the provisioned files via the Admin API).

Reviewer: ```suggestion
1. Restart your Grafana instance (or reload the provisioned files using the Admin API).
```

Here is an example of a configuration file for creating templates.

```yaml
# config file version
apiVersion: 1

# List of templates to import or update
templates:
  # <int> organization ID, default = 1
  - orgId: 1
    # <string, required> name of the template, must be unique
    name: my_first_template
    # <string, required> content of the the template
    template: |
      {{ define "my_first_template" }}
        Custom notification message
      {{ end }}
```

Here is an example of a configuration file for deleting templates.

```yaml
# config file version
apiVersion: 1

# List of alert rule UIDs that should be deleted
deleteTemplates:
  # <int> organization ID, default = 1
  - orgId: 1
    # <string, required> name of the template, must be unique
    name: my_first_template
```

## Import notification policies

> Create or reset the notification policy tree via provisioning files in your Grafana instance(s).

Reviewer: ```suggestion
Create or reset the notification policy tree using provisioning files in your Grafana instance(s).
```

{{< /admonition >}}

1. Find the notification policy tree in Grafana.
1. [Export][alerting_export] and download a provisioning file for your notification policy tree.

> 1. Copy the contents into a YAML or JSON configuration file and place it in the `provisioning/alerting` directory of the Grafana instance you wish to import the alerting resources.

Reviewer: ```suggestion
1. Copy the contents into a YAML or JSON configuration file and add it to the `provisioning/alerting` directory of the Grafana instance you want to import the alerting resources to.
```

   Example configuration files can be found below.

> 1. Restart your Grafana instance (or reload the provisioned files via the Admin API).

Reviewer: ```suggestion
1. Restart your Grafana instance (or reload the provisioned files using the Admin API).
```

  - 1
```

## Import mute timings

> Create or delete mute timings via provisioning files via provisioning files in your Grafana instance(s).

Reviewer: ```suggestion
Create or delete mute timings via provisioning files using provisioning files in your Grafana instance(s).
```

1. Find the mute timing in Grafana.
1. [Export][alerting_export] and download a provisioning file for your mute timing.

> 1. Copy the contents into a YAML or JSON configuration file and place it in the `provisioning/alerting` directory of the Grafana instance you wish to import the alerting resources.

Reviewer: ```suggestion
1. Copy the contents into a YAML or JSON configuration file and add it to the `provisioning/alerting` directory of the Grafana instance you want to import the alerting resources to.
```

   Example configuration files can be found below.

> 1. Restart your Grafana instance (or reload the provisioned files via the Admin API).

Reviewer: ```suggestion
1. Restart your Grafana instance (or reload the provisioned files using the Admin API).
```

         template: the content for my template
   ```

> 1. Restart your Grafana instance (or reload the provisioned files via the Admin API).

Reviewer: ```suggestion
1. Restart your Grafana instance (or reload the provisioned files using the Admin API).
```
